src/mark.ts

```diff
@@ -39,16 +38,16 @@ export default function mark(
         throw Error('Unsupported rule type for rule `' + rule + '`.');
     }
 
-    const output = [];
+    const output:React.ReactNodeArray = [];
 
     for (let part of content) {
         if (typeof part === 'string') {
-            let marked;
+            let marked: React.ReactNodeArray;
 
             if (rule instanceof RegExp) {
                 marked = markRegExp(part, rule, tag, matchIndex);
             }
-            else if (typeof rule === 'string') {
```

Comment:
Because this is a library and we can't anticipate how users are going to break it, I'd rather keep this check, and add a `raise` below in all other cases to explain that `rule` can only be a string or a regex.

Reply:
That input validation is already in line 37.

Reply:
Indeed! :+1: